clients/src/main/java/org/apache/kafka/clients/admin/MembershipChangeResult.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.admin;
+
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+import java.util.concurrent.ExecutionException;
+
+/**
+ * The result of the {@link KafkaAdminClient#removeMemberFromGroup(String, RemoveMemberFromGroupOptions)} call.
+ *
+ * The API of this class is evolving, see {@link AdminClient} for details.
+ */
+@InterfaceStability.Evolving
+public class MembershipChangeResult {
+
+    private KafkaFuture<RemoveMemberFromGroupResult> future;
```

Comment:
I'd suggest we organize the future differently to be consistent with others:

```
MembershipChangeResult: Map<MemberIdentity, KafkaFuture<Void>>
```

If the response has an error, that error should be thrown from the future.get() directly of the top-level future. Then within the map, we set the `future.complete` or `future.completeExceptionally` accordingly. Please see my other comment above.

Reply:
I'm not sure this would be right because the current API will only call leave group once, so it doesn't make sense to have separate futures. The cases like `deleteGroup`, `describeGroup` or `listGroup` are directing requests to different coordinators so they need to separate the futures for different group.id, while request like `ListConsumerGroupOffsetsResult` aims only at one group, and return only one future.

Reply:
Discussed with @abbccdda a bit offline, the main reason is to 1) align with the response / per-member error code hierarchy, and also 2) be a bit consistent with other return types (though I agree that in this case, all futures would be completed at the same time since they are from the same coordinator).